pkg/utils/files_test.go

```diff
@@ -0,0 +1,246 @@
+			file: "test",
+		},
+		want: false,
+	}}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := IsYaml(tt.args.file); got != tt.want {
+				t.Errorf("IsYaml() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestIsJson(t *testing.T) {
+	type args struct {
+		file string
+	}
+	tests := []struct {
+		name string
+		args args
+		want bool
+	}{{
+		name: "empty",
+		args: args{
+			file: "",
+		},
+		want: false,
+	}, {
+		name: "with yaml",
+		args: args{
+			file: "test.yaml",
+		},
+		want: false,
+	}, {
+		name: "with yml",
+		args: args{
+			file: "test.yml",
+		},
+		want: false,
+	}, {
+		name: "with json",
+		args: args{
+			file: "test.json",
+		},
+		want: true,
+	}, {
+		name: "with pdf",
+		args: args{
+			file: "test.pdf",
+		},
+		want: false,
+	}, {
+		name: "without extension",
+		args: args{
+			file: "test",
+		},
+		want: false,
+	}}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := IsJson(tt.args.file); got != tt.want {
+				t.Errorf("IsJson() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestIsYamlOrJson(t *testing.T) {
+	type args struct {
+		file string
+	}
+	tests := []struct {
+		name string
+		args args
+		want bool
+	}{{
+		name: "empty",
+		args: args{
+			file: "",
+		},
+		want: false,
+	}, {
+		name: "with yaml",
+		args: args{
+			file: "test.yaml",
+		},
+		want: true,
+	}, {
+		name: "with yml",
+		args: args{
+			file: "test.yml",
+		},
+		want: true,
+	}, {
+		name: "with json",
+		args: args{
+			file: "test.json",
+		},
+		want: true,
+	}, {
+		name: "with pdf",
+		args: args{
+			file: "test.pdf",
+		},
+		want: false,
+	}, {
+		name: "without extension",
+		args: args{
+			file: "test",
+		},
+		want: false,
+	}}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := IsYamlOrJson(tt.args.file); got != tt.want {
+				t.Errorf("IsYamlOrJson() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestFindFiles(t *testing.T) {
+	type args struct {
+		args []string
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    []string
+		wantErr bool
+	}{{
+		name: "one folder",
+		args: args{
+			args: []string{
+				"./testdata/a",
+			},
+		},
+		wantErr: false,
+		want: []string{
+			"testdata/a/a.json",
+			"testdata/a/a.yaml",
+			"testdata/a/a.yml",
+		},
+	}, {
+		name: "two folders",
+		args: args{
+			args: []string{
+				"./testdata/a",
+				"./testdata/b",
+			},
+		},
+		wantErr: false,
+		want: []string{
+			"testdata/a/a.json",
+			"testdata/a/a.yaml",
+			"testdata/a/a.yml",
+			"testdata/b/b.json",
+			"testdata/b/b.yaml",
+			"testdata/b/b.yml",
+		},
+	}, {
+		name: "recursive",
+		args: args{
+			args: []string{
+				"./testdata",
+			},
+		},
+		wantErr: false,
+		want: []string{
+			"testdata/a/a.json",
+			"testdata/a/a.yaml",
+			"testdata/a/a.yml",
+			"testdata/b/b.json",
+			"testdata/b/b.yaml",
+			"testdata/b/b.yml",
+		},
+	}, {
+		name: "invalid folder",
```

Comment:
That test was super useful and made me think a little more. I think asking explicitely for a thing that doesn't exist should be an error, asking for something that ends-up being empty should return nothing. What about asking specifically for a file that is not yaml? iow, what about that test:
```
args: []string{"testdata/a/a.pdf"}
```
what do you expect this to do? I would expect this to load the file.

Reply:
IMHO it should return nothing. Passing an argument that does not exist is an error, but passing a file that does not satisfy the predicate is the same as asking all files in a folder that satisfy the predicate.

You give a path in input (folder or file) and the func returns all files that match the `IsJsonOrYaml` predicate.

Reply:
From a UX perspective, if a user directly passes in a `.txt` or file with no extension to the CLI, I think their expectation is for it to be validated unconditionally. 

I think the filtering should apply only to folders, at least from the user's perspective on the CLI. FindFiles can be modified for this functionality, or its usage. I do like that the function has a simple description so perhaps we add all directly specified files from the user input to its result.

Reply:
@alexzielenski 👍

Reply:
Done.